File: problems/algorithms/object-order-item-oo/js/response/product.js
class Product {
  constructor(id, name, price, categories) {
    this.id = id;
    this.name = name;
    this.price = price || 0.0;
    this.categories = categories || [];
  }

  getNameCompact() {
    return this.name.substr(0, 10).padEnd(10);
  }
}

export { Product };

File: problems/algorithms/object-order-item-oo/ruby/response/product.rb
class Product
    def initialize(id, name, price, categories = [])
        @id = id
        @name = name
        @price = price || 0.0
        @categories = categories
    end

    def get_name_compact
        return @name[0, 10].ljust(10, " ")
    end

    def id
        @id
    end

    def name
        @name
    end

    def price
        @price
    end

    def categories
        @categories
    end

    def to_s
        "[Produto - Nome: #{@name}, Preço: #{@price}]"
    end
end
